include/linux/dmar.h

```diff
@@ -45,7 +45,7 @@ struct dmar_drhd_unit {
 	u64	reg_base_addr;		/* register base address*/
 	struct	dmar_dev_scope *devices;/* target device array	*/
 	int	devices_cnt;		/* target device count	*/
-	u16	segment;		/* PCI domain		*/
+	int	segment;		/* PCI domain		*/
```

Comment:
You are not only changing the size here, but also changing an unsigned type to a signed one -- shouldn't this be u32 instead? (same applies for all other u16 -> int changes on the same commit)